Update navbar

nikoshell opened the pull request.

nikoshell committed:
Update navbar

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

europython-society commented:
# Preview available
|Key|Value|
|---|---|
|url|https://ep2025-navbar.ep-preview.click|
|last update|2025-06-17T16:03:16.635Z|

clytaemnestra commented:
Discussed in a DM - it still doesn't work. I'm able to reproduce this issue on iPhone 15, in Safari and Chrome.

nikoshell committed:
Remove focus from navbar

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

nikoshell committed:
Merge branch 'ep2025' into ep2025-navbar

nikoshell commented:
> Discussed in a DM - it still doesn't work. I'm able to reproduce this issue on iPhone 15, in Safari and Chrome.

Could you retest new version?

clytaemnestra committed:
Merge branch 'ep2025' into ep2025-navbar

clytaemnestra commented:
it doesn't work with the newest version either 

![image](https://github.com/user-attachments/assets/b531f5a0-d0fd-44b6-a836-9d4d33a889f5)

nikoshell committed:
Update navbar script

Debouncing with requestAnimationFrame
Minimum scroll threshold
Top/bottom boundary handling
State tracking
Passive scroll listeners
Orientation change handling
Visibility API handling
Smooth transitions

nikoshell committed:
Fix key press

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

nikoshell committed:
Fix bounce

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

nikoshell committed:
Merge branch 'ep2025' into ep2025-navbar

clytaemnestra commented:
it works for me 🚀🚀🚀 

![image](https://github.com/user-attachments/assets/fa3072d3-cc93-48d5-a112-adda8c652b75)
